File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        major = nums[0]
        cnt = 1
        for i in range(1,len(nums)):
            if(cnt==0):
                major = nums[i]
            if nums[i]==major:
                cnt+=1
            else:
                cnt-=1
        return major
        """
        :type nums: List[int]
        :rtype: int
        """
        

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int cnt = 1, num = nums[0];
        for(int i=1;i<n;i++){
            if(cnt==0){
                num = nums[i];
            }
            if(nums[i]==num){
                cnt++;
            }else{
                cnt--;
            }
        }
        return num;
    }
};